src/components/environment-picker.tsx

```diff
@@ -32,20 +32,39 @@ export function EnvironmentPicker(props: EnvironmentPickerProps): JSX.Element {
   return (
     <FormControl>
       <InputLabel id={labelId}>{props.label}</InputLabel>
-      <Select
-        labelId={labelId}
-        label={props.label}
-        name={props.name}
-        id={props.id}
-        onChange={props.onChange}
-        value={props.initialValue}
-      >
-        {props.environmentList.map((env, idx) => (
-          <MenuItem value={env.label} title={env.description} key={idx}>
-            {env.name}
+      {props.environmentList.length === 1 ? (
```

Comment:
If the environment list has exactly one item in it, then we should:

1. Display a read-only text field with the one item in it
2. Display the output formats for the one environment
3. Display the compute types for the one environment
4. Save the one environment in memory so that, when the user submits the form, the one environment is present in the API call

Reply:
A text field will be confusing to the user none the less. We should rather not display the environment field at all if there is a single value. To make this all work within the form, you can do one of these 2 options:
1. Directly set the default value on the model and everything else works? I am not sure if it will.
2. Use a hidden field `<input type="hidden" value="environment_a">` instead of the environments dropdown.

Output formats, compute types should mostly pick up the value from the model and work automagically?